tex/generic/pgf/systemlayer/pgfsys-dvips.def

```diff
@@ -159,28 +164,491 @@
+      pgfcheckcolorrange 16777215 mul round cvi % -str- sampleindex gray24
+      exch pgfsamplesizegray mul exch
+      2 index 2 index       2 index           -16 bitshift put % high byte
+      2 index 2 index 1 add 2 index 65535 and  -8 bitshift put % middle byte
+      2 index 2 index 2 add 2 index   255 and              put % low byte
+      pop pop pop
+    } bind def
+    %
+    % <colour component> pgfcheckcolorrange ==> <something between 0.0 and 1.0>
+    % truncates value to the allowed range (user-defined Type-4 functions may
+    % happen to provide values outside this range)
+    /pgfcheckcolorrange {
+      dup 0.0 ge not {pop 0.0} if
+      dup 1.0 le not {pop 1.0} if
+    } bind def
+    %
+    %colour depths per channel (bit)
+    /pgfchanneldepthcmyk 8 def
+    /pgfchanneldepthrgb 8 def
+    /pgfchanneldepthgray 24 def
+    %
+    %colour sample size (byte)
+    /pgfsamplesizecmyk 4 def
+    /pgfsamplesizergb 3 def
+    /pgfsamplesizegray 3 def
+    %
+    %/Range array in the FunctionType 0 dictionary
+    /pgfrangecmyk [0 1 0 1 0 1 0 1] def
+    /pgfrangergb [0 1 0 1 0 1] def
+    /pgfrangegray [0 1] def
+  }%
+}
+
+
+%
+% Opacity masks
+%
+
+\def\pgfsys@fadingfrombox#1#2{%
+  \ifcsname pgfsmaskxform@#1\endcsname
+    \pgfwarning{Fading '#1' already defined.\MessageBreak Going to replace it with new definition}%
+  \fi%
+  \global\advance\pgf@objectcount\@ne%
+  {%
+    \pgf@x=-.5\wd#2%
+    \pgf@y=-.5\ht#2%
+    \advance\pgf@y by.5\dp#2%
+    \expandafter\xdef\csname pgfsmasktrans@#1\endcsname{%
+      \noexpand\pgftransformcm{1}{0}{0}{1}{\noexpand\pgfqpoint{\the\pgf@x}{\the\pgf@y}}}%
+    \edef\@tempa{\noexpand\pgfutil@insertatbegincurrentpagefrombox{%
+      % box coordinates (reference point and top) needed for distilling
+      \pgfsys@outerinvoke{currentpoint /pgf@refy exch def /pgf@refx exch def}%
+      \hbox to 0pt {\hbox to \ifdim\ht#2>\ht\strutbox\ht#2\else\ht\strutbox\fi{\hss\pgfsys@outerinvoke{%
+        /pgf@top pgf@refy currentpoint pop pgf@refx sub VResolution Resolution div mul sub def%
+      }}\hss}%
+      \pgfsys@outerinvoke{gsave
+        % translate box to upper left page corner, so we have the whole clipping path (i. e.
+        % page area) available for distilling, as outlying parts get clipped
+        clippath pathbbox newpath pop pop translate pgf@refx neg pgf@top neg translate
+        % translate origin (0,0) to the reference point
+        gsave pgf@refx pgf@refy translate
+        mark /_objdef {pgfsmaskxform@\the\pgf@objectcount}
+          /BBox [clippath pathbbox newpath] /BP pdfmark grestore%
+      }%
+      \wd#2=0pt\ht#2=0pt\dp#2=0pt%
+      \box#2%
+      \pgfsys@outerinvoke{mark /EP pdfmark grestore}%
+    }}\@tempa%
+    \expandafter\xdef\csname pgfsmaskxform@#1\endcsname{\the\pgf@objectcount}%
+  }%
+}
+\def\pgfsys@usefading#1#2#3#4#5#6#7{%
+  \ifcsname pgfsmaskxform@#1\endcsname%
+  {%
+     \pgftransformreset%
+     \pgftransformcm{#2}{#3}{#4}{#5}{\pgfpoint{#6}{#7}}%
```

Comment:
Do the other drivers handle non-linear transformations?  If yes, this one should too.

Reply:
Could you please elaborate on this, as I don't quite understand. The code is very much in line with that of the other drivers.

Reply:
Then it's fine.